src/utils/outputs/style-dictionary/colors.ts

```diff
@@ -0,0 +1,157 @@
+import { DefaultColorType, HeisslufType, SpeakingName } from "../../data.ts";
+import { getPalette } from "../index.ts";
+import { setObjectByPath } from "./index.ts";
+
+export const getSDColorPalette = (
+  allColors: Record<string, DefaultColorType>,
+  luminanceSteps: number[],
+): any => {
+  const palette: Record<string, HeisslufType[]> = getPalette(
+    allColors,
+    luminanceSteps,
+  );
+  const colors: any = {};
+
+  Object.entries(allColors).forEach(([unformattedName, color]) => {
+    const name = unformattedName.toLowerCase();
+
+    const colorValues: any = {};
+
+    const hslType: HeisslufType[] = palette[unformattedName];
+    hslType.forEach((hsl) => {
+      colorValues[`${hsl.index ?? hsl.name}`] = { value: hsl.hex };
+    });
+
+    colorValues.origin = {
+      base: {
+        comment: "This is just to resolve the original origin color",
+        value: color.origin,
+      },
+    };
+
+    colorValues.light = {
+      origin: {
+        default: {
+          value: color.originLight,
+        },
+        hovered: {
+          value: color.originLightHovered,
+        },
+        pressed: {
+          value: color.originLightPressed,
+        },
+      },
+      on: {
+        origin: {
+          default: {
+            value: color.onOriginLight,
+          },
+          hovered: {
+            value: color.onOriginLightHovered,
+          },
+          pressed: {
+            value: color.onOriginLightPressed,
+          },
+        },
+      },
+    };
+
+    colorValues.dark = {
+      origin: {
+        default: {
+          value: color.originDark,
+        },
+        hovered: {
+          value: color.originDarkHovered,
+        },
+        pressed: {
+          value: color.originDarkPressed,
+        },
+      },
+      on: {
+        origin: {
+          default: {
+            value: color.onOriginDark,
+          },
+          hovered: {
+            value: color.onOriginDarkHovered,
+          },
+          pressed: {
+            value: color.onOriginDarkPressed,
+          },
+        },
+      },
+    };
+
+    colors[name] = colorValues;
+  });
+
+  return { colors };
+};
+
+export const getSDSpeakingColors = (
+  speakingNames: SpeakingName[],
+  allColors: Record<string, DefaultColorType>,
+): any => {
+  const colors: any = { light: {}, dark: {} };
+  const colorTheme = ["light", "dark"];
+  for (const [unformattedName] of Object.entries(allColors)) {
+    const name = unformattedName.toLowerCase();
+
+    for (const theme of colorTheme) {
+      const isDark = theme === "dark";
+      const themeObj: any = {
+        origin: {
+          default: {
+            value: `{colors.${name}.${theme}.origin.default.value}`,
```

Comment:
**issue**: I missed an error in my previous suggestion. References do not include the `.value` part at the end.

```suggestion
            value: `{colors.${name}.${theme}.origin.default}`,
```

Reply:
nevermind - thats incorrect.